pyrs/core/summary_generator_stress.py

```diff
@@ -0,0 +1,200 @@
+                            entries += _write_number(d_value) + _write_number(d_error)
+
+                        elif field == 'FWHM':
+                            fwhm_value = peak_collection.get_effective_params()[0]['FWHM'][row]
+                            fwhm_error = peak_collection.get_effective_params()[1]['FWHM'][row]
+                            entries += _write_number(fwhm_value) + _write_number(fwhm_error)
+
+                        elif field == 'Peak_Height':
+                            height_value = peak_collection.get_effective_params()[0]['Height'][row]
+                            height_error = peak_collection.get_effective_params()[1]['Height'][row]
+                            entries += _write_number(height_value) + _write_number(height_error)
+
+                return entries
+
+            # Function starts here
+            body = ''
+            for row, coordinate in enumerate(self._stress.coordinates):
+
+                line = str(coordinate[0]) + ', ' + str(coordinate[1]) + ', ' + str(coordinate[2]) + ', '
+
+                # d0 doesn't depend on direction so just picking the first peak_collection
+                for direction in SummaryGeneratorStress.directions:
+                    peak_collection = self._get_peak_collection(direction)
+                    if not peak_collection:
+                        continue
+                    d0_value = peak_collection.get_d_reference()[0][row]
+                    d0_error = peak_collection.get_d_reference()[1][row]
+
+                    line += _write_number(d0_value) + _write_number(d0_error)
+                    break
+
+                # fields_3dir = ['d', 'FWHM', 'Peak_Height', 'Strain', 'Stress']
+                for field_3dir in SummaryGeneratorStress.fields_3dir:
+                    line += _write_field_3d(row, field_3dir)
+
+                line = line[:-2] + '\n'
+
+                body += line
+
+            handle.write(body)
+            return
+
+        # function starts here
+        with open(self._filename, 'w') as handle:
+            self._write_csv_header(handle)
+            _write_summary_csv_column_names(handle)
+            _write_summary_csv_body(handle)
+
+        return
+
+    def _get_strain_field(self, direction: str) -> Optional[StrainField]:
+        """
+            Returns a StrainField for a particular direction from self._stress
+        """
+        stress: StressField = self._stress
+        stress.select(direction)
+        strain = self._stress.strain
+        if isinstance(strain, StrainField):
+            return strain
+
+        return None
+
+    def _get_peak_collection(self, direction: str) -> Optional[PeakCollection]:
+        """
+            Returns a peak_collection for a particular direction from self._stress
+        """
+        strain = self._get_strain_field(direction)
+        if isinstance(strain, StrainField):
+            return strain.peak_collection
```

Comment:
This will raise a `RuntimeError` when the strain is made up of more than one peak collection. See here:
https://github.com/neutrons/PyRS/blob/master/pyrs/dataobjects/fields.py#L750
If you only want one, then `return strain.peak_collections[0]`. This will also work with strains containing only one peak collection